src/utils/getSearchWith.ts

```diff
@@ -0,0 +1,21 @@
+import { Params } from '../types';
+
+export function getSearchWith(
+  params: Params,
+  search?: string | URLSearchParams,
+) {
```

Comment:
The `search` parameter is optional and can be either a string or an instance of URLSearchParams. This could potentially cause confusion for other developers. Consider adding a comment to explain how this parameter is used.